Assets/00_Content/Scripts/Vikings/VikingController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using UnityEngine;
+
+namespace Vikings {
+	public class VikingController : MonoBehaviour {
+		[SerializeField] private Viking vikingPrefab;
+		[SerializeField] private Transform[] spawnPoints;
+		[SerializeField] private float spawnDelay = 2.5f;
+
+		private Viking[] spawnedVikings;
+		private float spawnTimer;
+
+		public bool CanSpawn { get; set; } = true;
+
+		private void Start() {
+			spawnedVikings = new Viking[spawnPoints.Length];
+		}
+
+		private void Update() {
+			if (!CanSpawn) return;
+
+			spawnTimer = Mathf.Max(0, spawnTimer - Time.deltaTime);
+
+			if (spawnTimer <= 0) {
+				SpawnViking();
+				spawnTimer = spawnDelay;
+			}
+		}
+
+		private void SpawnViking() {
+			int index = GetSpawnIndex();
+
+			if (index == -1) return;
+
+			Vector3 position = spawnPoints[index].position;
+			Viking viking = Instantiate(vikingPrefab, position, Quaternion.identity, transform);
+
+			spawnedVikings[index] = viking;
+
+			viking.LeaveTavern += OnLeaveTavern;
```

Comment:
Does the Viking Controller keep a check of how many vikings there are in the Tavern then? Because in that case  should I maybe remove that functionality from the tavern then?

Reply:
I think there has been a bit of a misunderstanding here, either place could work but I saw the tavern/round controller more as the one sending data on how the viking controller should act. This also depends greatly on if we have an actual limit on vikings in the tavern or not.

Reply:
Yeah, I think it is better if the VikingManager handles it tbh.

Especially seeing as the tavern should not care for the amount of guests. However, this raises the question of if the players can even accept more guests when there are no seating free?